metrics/core/src/main/java/org/hyperledger/besu/metrics/prometheus/PrometheusMetricsSystem.java

```diff
@@ -280,11 +298,36 @@ public String convertToPrometheusName(final MetricCategory category, final Strin
     return prometheusPrefix(category) + name;
   }
 
+  /**
+   * Convert to prometheus counter name. Prometheus adds a _total suffix to the name if not present,
+   * so we remember if the original name already has it, to be able to covert back correctly
+   *
+   * @param category the category
+   * @param name the name
+   * @return the name as string
+   */
+  public String convertToPrometheusCounterName(final MetricCategory category, final String name) {
+    if (name.endsWith("_total")) {
+      totalSuffixedCounters.add(name);
+    }
+    return convertToPrometheusName(category, name);
+  }
+
   private String convertFromPrometheusName(final MetricCategory category, final String metricName) {
     final String prefix = prometheusPrefix(category);
     return metricName.startsWith(prefix) ? metricName.substring(prefix.length()) : metricName;
   }
 
+  private String convertFromPrometheusCounterName(
+      final MetricCategory category, final String metricName) {
+    final String prefix = prometheusPrefix(category);
+    final String unPrefixedName =
+        metricName.startsWith(prefix) ? metricName.substring(prefix.length()) : metricName;
+    return totalSuffixedCounters.contains(unPrefixedName + "_total")
+        ? unPrefixedName + "_total"
```

Comment:
I understand this could be a little misleading, but internally Prometheus stores the counter name without the `_total` suffix, and if a name already has the suffix, it is removed, so when converting back from Prometheus to Besu names, if we remember that the name already had the suffix, we need to readd it.